File: sdi3-14.Cli-REST/src/uo/sdi/clients/rest/menu/items/LoginMenuItem.java
package uo.sdi.clients.rest.menu.items;

import java.util.ArrayList;
import java.util.List;

import uo.sdi.clients.rest.menu.MenuItem;
import uo.sdi.clients.rest.restService.RestService;
import alb.util.console.Console;

public class LoginMenuItem implements MenuItem {

	private RestService restService;

	public LoginMenuItem() {
	}

	@Override
	public String getMenuName() {
		return "Iniciar sesion";
	}

	@Override
	public List<MenuItem> execute() {
		List<MenuItem> items = new ArrayList<MenuItem>();

		String login = Console.readString("Introduce tu login:");
		String password = Console.readString("Introduce tu contrasenia");

		restService = new RestService(login, password);
		
		Long user = restService.login();

		if (user != null) {
			items.add(new ListCategoriesMenuItem(restService));
			items.add(new ListTaskFromCategoryMenuItem(restService));
			items.add(new CreateTaskMenuItem(restService));
			items.add(new MarkTaskAsFinishedMenuItem(restService));
			items.add(new LogoutMenuItem());
			
			System.out.println("Login correcto");
			
		} else {
			items.add(this);
			System.out.println("Login incorrecto o servidor no accesible");
		}

		return items;
	}

}

File: sdi3-14.Cli-REST-CSharp/sdi3-14.Cli-REST-CSharp/menu/menuItems/LoginMenuItem.cs
using sdi3_14.Cli_REST_CSharp.dto;
using sdi3_14.Cli_REST_CSharp.restService;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_14.Cli_REST_CSharp.menu.menuItems
{
    class LoginMenuItem : MenuItem
    {
        private RestService restService;

        public LoginMenuItem()
        {
        }

        public String GetMenuName()
        {
            return "Iniciar sesion";
        }

        public IList<MenuItem> Execute()
        {
            List<MenuItem> items = new List<MenuItem>();

            Console.Write("Introduce tu login:");
            String login = Console.ReadLine();

            Console.Write("Introduce tu contrasenia:");
            String password = Console.ReadLine();

            restService = new RestService(login, password);

            Int64? user = restService.login();

            if (user != null)
            {
                items.Add(new ListCategoriesMenuItem(restService));
                items.Add(new ListTaskFromCategoryIdMenuItem(restService));
                items.Add(new CreateTaskMenuItem(restService));
                items.Add(new MarkTaskAsFinishedMenuItem(restService));
                items.Add(new LogoutMenuItem());

                Console.WriteLine("Login correcto");

            }
            else
            {
                items.Add(this);
                Console.WriteLine("Login incorrecto o servidor no accesible");
            }

            return items;
        }
    }
}
